pandas/core/indexes/numeric.py

```diff
@@ -165,7 +166,11 @@ def _ensure_array(cls, data, dtype, copy: bool):
         dtype = cls._ensure_dtype(dtype)
 
         if copy or not is_dtype_equal(data.dtype, dtype):
-            subarr = np.array(data, dtype=dtype, copy=copy)
+            try:
```

Comment:
why do you think you need this try/except? e.g. validate_dtype should raise correct?

Reply:
There are a lot of combinations of `data` and `dtype` and it's difficult (for me) to nail down to something sensible. For example:

```python
>>> data =  [pd.Timestamp("20130101")]
>>> NumericIndex(data)
>>> NumericIndex(data, dtype="float64")
>>> NumericIndex(np.array(data))
```

may not give the same error or error message.

I've tried again now and can't seem to find anything clearer. I'm not so happy about it myself, but there just seems to be very difficult to improve (for me, maybe someone else can find something better).

Reply:
i c, ok, can you comment on this (as we may want to try to figure this out in the future)

Reply:
Ok.

Reply:
can you add a "TODO" to this comment

Reply:
Ok.